pgmax/fg/fg_utils.py

```diff
@@ -1,50 +1,91 @@
 """A module containing helper functions for constructing Factor Graphs."""
 
-from typing import Sequence
+from typing import Sequence, Union
 
 import numpy as np
 
-from pgmax.fg.nodes import EnumerationWiring
+from pgmax.fg.nodes import EnumerationWiring, ORWiring
 
 
 def concatenate_enumeration_wirings(
-    wirings: Sequence[EnumerationWiring],
-) -> EnumerationWiring:
+    enum_wirings: Sequence[EnumerationWiring], start_edge_states: np.ndarray
```

Comment:
We can go with this approach and get to a complete, runnable implementation first, but this way of doing things might get annoying quickly.

If you just aggregate different types of factors into different containers (e.g. one list containing all enumeration factors, and another list containing all OR factors), you can then have one global offset which you apply to all factors. This also decouples things as much as possible.